docs/source/guide.rst

```diff
@@ -39,14 +39,136 @@ The hook will retrieve a balance of an ERC20 token of the provided address.
     return tokenBalance
   }
 
+Another example can be useTokenAllowance. Instead of balanceOf we use allowance on ERC20 interface.
+
+.. code-block:: javascript
+
+  function useTokenAllowance(
+    tokenAddress: string | Falsy,
+    ownerAddress: string | Falsy,
+    spenderAddress: string | Falsy
+  ) {
+    const [allowance] =
+      useContractCall(
+        ownerAddress &&
+          spenderAddress &&
+          tokenAddress && {
+            abi: ERC20Interface,
+            address: tokenAddress,
+            method: 'allowance',
+            args: [ownerAddress, spenderAddress],
+          }
+      ) ?? []
+    return allowance
+  }
+
+
 The *useContractCall* hook will take care of updating the balance of new blocks.
 The results are deferred so that the hook does not update too frequently.
 
+In our custom hooks we can use any standard react hooks, custom react hooks and useDApp hooks.
+Rules of hooks apply.
+
 All core hooks are available `here <https://github.com/EthWorks/useDapp/tree/master/packages/core/src/hooks>`_.
 
 
 Testing hooks
 *************
 
-Coming soon...
+Testing should be easy and straightforward. We just need a small setup and we are good to go.
```

Comment:
Opinion. Focus on facts. Better:
To write test, start with a setup code that will create a mock provider and  test wallets.